modules/custom_operations/user_ie_extensions/fft.cpp

```diff
@@ -0,0 +1,369 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "fft.hpp"
+
+#include <ie_common.h>
+#include <ie_parallel.hpp>
+#include <details/ie_so_loader.h>
+#include <opencv2/core/core_c.h>
+
+using namespace TemplateExtension;
+
+std::unique_ptr<InferenceEngine::details::SharedObjectLoader> so;
+using cvCreateMatHeaderF = CvMat*(int, int, int);
+using cvSetDataF = void(CvArr*, void*, int);
+using cvReleaseMatF = void(CvMat**);
+using cvDftF = void(const CvArr*, CvArr*, int, int);
+using cvScaleF = void(const CvArr*, CvArr*, double, double);
+using cvCloneMatF = CvMat*(const CvMat*);
+using cvCopyF = void(const CvArr*, const CvArr*, const CvArr*);
+using cvInitMatHeaderF = CvMat*(CvMat*, int, int, int, void*, int);
+using cvGetRawDataF = void(const CvArr*, uchar**, int* step, CvSize* roi_size);
+using cvReshapeF = CvMat*(const CvArr*, CvMat*, int, int);
+using cvCreateDataF = void(CvArr*);
+using cvReleaseDataF = void(CvArr*);
+
+bool loadOpenCV() {
+    static bool loaded = false;
+    if (!loaded) {
+        loaded = true;
+        try {
+#ifdef _WIN32
+            so.reset(new InferenceEngine::details::SharedObjectLoader("opencv_core.dll"));
+#elif defined(__APPLE__)
+            so.reset(new InferenceEngine::details::SharedObjectLoader("libopencv_core.dylib"));
+#else
+            so.reset(new InferenceEngine::details::SharedObjectLoader("libopencv_core.so"));
+#endif
```

Comment:
@ilyachur do you have suggestions on this? 
We will have to rework this place once we remove API 1.0.. Moreover, details namespace is not allowed for public users.

Reply:
I am not sure that we need to provide an API to load some custom runtime libraries, it is not a responsibility of OpenVINO toolkit.

Reply:
@ilya-lavrenov @ilyachur This means we can only test this code in CI with warnings-as-errors turned off. What would be the recommended approach?

Reply:
Reimplement cross-platform library loading directory in this library and not rely on OV deprecated API.

Reply:
You mean something like this? 

- https://github.com/maddouri/dynalo
- https://github.com/martin-olivier/dylib

Come to think of it, perhaps @dkurt knows why we are loading the OpenCV library at runtime, instead of just linking to it?

Reply:
Because we don't need extra OpenCV dependencies in some projects. That was a requirement to use OpenCV as an optional plugin.

Reply:
@dkurt what do you mean by "extra OpenCV dependencies" ?

Reply:
I mean without OpenCV installed nut linked, there will be an error "cannot find libopencv.so". As a plugin we can make this dependency optional.

Reply:
I see, in case someone wants to load the extension library on a system without OpenCV, but wants to use any of the operations that do not require OpenCV (like fft), even if fft is compiled into the extension library. 

On the other hand, I see this "contrib" extension library "reference code", not something someone would use immediately in production. If someone really doesn't want to include FFT (with OpenCV dependency), then just don't compile it into the extension library. We even have a way to do that with the `CUSTOM_OPERATIONS` CMake variable (a list of ops to include). 

I am wary of adding a bunch of code (or external dependency) just to show we *can* do this. Instead, I recommend making changes to the build that makes sure the extension library is conditionally linked to OpenCV only when FFT (or any other operation that requires OpenCV) is included in the build.